cmd/ctr/commands/tasks/ps.go

```diff
@@ -38,21 +36,19 @@ var psCommand = cli.Command{
 		if err != nil {
 			return err
 		}
-		w := tabwriter.NewWriter(os.Stdout, 10, 1, 3, ' ', 0)
+		w := tabwriter.NewWriter(os.Stdout, 1, 8, 4, ' ', 0)
 		fmt.Fprintln(w, "PID\tINFO")
 		for _, ps := range processes {
+			var info interface{} = "-"
 			if ps.Info != nil {
-				var details hcsshimtypes.ProcessDetails
-				if err := details.Unmarshal(ps.Info.Value); err == nil {
-					if _, err := fmt.Fprintf(w, "%d\t%+v\n", ps.Pid, details); err != nil {
-						return err
-					}
-				}
-			} else {
-				if _, err := fmt.Fprintf(w, "%d\t-\n", ps.Pid); err != nil {
+				info, err = unmarshalProcessInfo(ps.Info)
```

Comment:
If you use the `typeurl` package an UnmarshalAny method you don't have to have this unmarshalProcessInfo func, it will unmarshal into the correct type and you can just print it out as normal.